### ISSUE TITLE
yield from return a E901 invalid syntax

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#96]
In GitLab by @Natim on Apr 8, 2015, 05:18

```python
value = yield from redis.blpop("channel")
```

```
$ flake8 test.py 
test.py:1:19: E901 SyntaxError: invalid syntax
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 8, 2015, 05:42

Is flake8 installed on a version of Python 3 where yield from is valid syntax?

**Comment by asottile** [comment]
In GitLab by @Natim on Apr 8, 2015, 05:49

Apparently yes:

```
(python34_venv) ~/$ pip install flake8
Requirement already satisfied (use --upgrade to upgrade): flake8 in ~/.virtualenvs/python34_venv/lib/python3.4/site-packages
Cleaning up...
(python34_venv) ~/$ flake8 test.py 
test.py:1:19: E901 SyntaxError: invalid syntax

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 8, 2015, 05:53

Can you show the output of `python --version` and `which flake8`.

**Comment by asottile** [comment]
In GitLab by @Natim on Apr 8, 2015, 06:05

```
(python34_venv) ~/$ which flake8 
/home/rhubscher/.virtualenvs/python34_venv/bin/flake8
(python34_venv) ~/$ which python
/home/rhubscher/.virtualenvs/python34_venv/bin/python
(python34_venv) :~/$ python --version
Python 3.4.0
```

**Comment by asottile** [comment]
In GitLab by @Natim on Apr 8, 2015, 06:13

Apparently pep8 doesn't raise this error neither does pyflakes

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 8, 2015, 06:15

```
(flake8-53)~/v/flake8-53 ❯❯❯ vim test.py
(flake8-53)~/v/flake8-53 ❯❯❯ python test.py
  File "test.py", line 1
    value = yield from (1, 2, 3)
           ^
SyntaxError: 'yield' outside function
```

Python does raise that error. `yield from` needs to be inside a function.

**Comment by asottile** [comment]
In GitLab by @Natim on Apr 8, 2015, 06:31

This was just a simple example to reproduce the problem, it is the same with:

```
import asyncio


@asyncio.coroutine
def test():
    value = yield from (1, 2, 3)
    return value
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 8, 2015, 06:35

```
(flake8-53)~/v/flake8-53 ❯❯❯ python test.py
(flake8-53)~/v/flake8-53 ❯❯❯ bin/flake8 test.py
(flake8-53)~/v/flake8-53 ❯❯❯ bin/flake8 --version
2.4.0 (pep8: 1.5.7, pyflakes: 0.8.1, mccabe: 0.3) CPython 3.4.2 on Darwin
(flake8-53)~/v/flake8-53 ❯❯❯ cat test.py
import asyncio


@asyncio.coroutine
def test():
    value = yield from (1, 2, 3)
    return value
```

I can't reproduce this in the slightest.

**Comment by asottile** [comment]
In GitLab by @Natim on Apr 8, 2015, 06:39

```
$ flake8 --version
2.1.0 (pep8: 1.4.6, mccabe: 0.2.1, pyflakes: 0.7.3) CPython 2.7.6 on Linux
```

That might be the problem.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 8, 2015, 06:40

As I said, flake8 needs to be installed on the right version of Python. There's nothing we can do to prevent syntax errors on Python 2.7 for syntax only valid on 3.x.

**Comment by asottile** [comment]
In GitLab by @Natim on Apr 8, 2015, 06:41

Ok thank you for your help on this.
